
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    <title>Examples &#8212; ENCODE Utils 0.3.0 documentation</title>
    <link rel="stylesheet" href="_static/alabaster.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    './',
        VERSION:     '0.3.0',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true,
        SOURCELINK_SUFFIX: '.txt'
      };
    </script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="eu_register.py" href="scripts/eu_register.html" />
    <link rel="prev" title="ENCODE Utils package" href="index.html" />
   
  <link rel="stylesheet" href="_static/custom.css" type="text/css" />
  
  
  <meta name="viewport" content="width=device-width, initial-scale=0.9, maximum-scale=0.9" />

  </head>
  <body>
  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <div class="section" id="examples">
<h1>Examples<a class="headerlink" href="#examples" title="Permalink to this headline">¶</a></h1>
<div class="section" id="imports">
<h2>Imports<a class="headerlink" href="#imports" title="Permalink to this headline">¶</a></h2>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="kn">import</span> <span class="nn">encode_utils</span> <span class="k">as</span> <span class="nn">eu</span>
<span class="kn">from</span> <span class="nn">encode_utils.connection</span> <span class="k">import</span> <span class="n">Connection</span>
</pre></div>
</div>
</div>
<div class="section" id="connecting-to-the-production-and-development-portals">
<h2>Connecting to the production and development Portals<a class="headerlink" href="#connecting-to-the-production-and-development-portals" title="Permalink to this headline">¶</a></h2>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="c1">#prod portal:</span>
<span class="n">conn</span> <span class="o">=</span> <span class="n">Connection</span><span class="p">(</span><span class="s2">&quot;prod&quot;</span><span class="p">)</span>

<span class="c1">#dev portal</span>
<span class="n">conn</span> <span class="o">=</span> <span class="n">Connection</span><span class="p">(</span><span class="s2">&quot;dev&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="section" id="get-request">
<h2>GET Request<a class="headerlink" href="#get-request" title="Permalink to this headline">¶</a></h2>
<p>Retrieve the JSON serialization for the Experiment record with accession ENCSR161EAA:</p>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="n">conn</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;ENCSR161EAA&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="section" id="search">
<h2>Search<a class="headerlink" href="#search" title="Permalink to this headline">¶</a></h2>
<p>Search for ChIP-seq assays performed on primary cells from blood:</p>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="n">query</span> <span class="o">=</span> <span class="p">{</span>
  <span class="s2">&quot;assay_title&quot;</span><span class="p">:</span> <span class="s2">&quot;ChIP-seq&quot;</span><span class="p">,</span>
  <span class="s2">&quot;biosample_type&quot;</span><span class="p">:</span> <span class="s2">&quot;primary cell&quot;</span><span class="p">,</span>
  <span class="s2">&quot;organ_slims&quot;</span><span class="p">:</span> <span class="s2">&quot;blood&quot;</span><span class="p">,</span>
  <span class="s2">&quot;type&quot;</span><span class="p">:</span> <span class="s2">&quot;Experiment&quot;</span>
<span class="p">}</span>

<span class="n">conn</span><span class="o">.</span><span class="n">search</span><span class="p">(</span><span class="n">query</span><span class="p">)</span>
</pre></div>
</div>
<p>The query will be URL encoded for you.  If you want to use the search functionality
programmatically, you should first test your search interactively on the Portal. The result will
be an array of record results, where each result is given in its JSON representation.</p>
</div>
<div class="section" id="patch-request">
<h2>PATCH Request<a class="headerlink" href="#patch-request" title="Permalink to this headline">¶</a></h2>
<p>Add a new alias to the GeneticModification record ENCGM063ASY. Create a payload
(<cite>dict</cite>) that indicates the record to PATCH and the new alias. The record to PATCH must be
indicated by using the non-schematic key <cite>Connection.ENCID_KEY</cite>, or <cite>self.ENCID_KEY</cite> from the
perspective of a <cite>Connection</cite> instance, which will be removed from the payload prior to submission:</p>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="n">payload</span> <span class="o">=</span> <span class="p">{</span>
  <span class="n">conn</span><span class="o">.</span><span class="n">ENCID_KEY</span><span class="p">:</span> <span class="s2">&quot;ENCGM063ASY&quot;</span><span class="p">,</span>
  <span class="s2">&quot;aliases&quot;</span><span class="p">:</span> <span class="p">[</span><span class="s2">&quot;new-alias&quot;</span><span class="p">]</span>
  <span class="p">}</span>

<span class="n">conn</span><span class="o">.</span><span class="n">patch</span><span class="p">(</span><span class="n">payload</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="section" id="file-upload">
<h2>File Upload<a class="headerlink" href="#file-upload" title="Permalink to this headline">¶</a></h2>
<p>Given the File record ENCFF852WVP, say that we need to upload the corresponding FASTQ file to AWS
(i.e. maybe the former attempt failed at the time of creating the File record). Here’s how to
do it:</p>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="n">conn</span><span class="o">.</span><span class="n">upload_file</span><span class="p">(</span><span class="n">file_id</span><span class="o">=</span><span class="s2">&quot;ENCFF852WVP&quot;</span><span class="p">)</span>
</pre></div>
</div>
<p>This will only work if the File record has the <cite>submitted_file_name</cite> property set, which is
interperted as the local path to the file to submit.
You can also explicitely set the path to the file to upload:</p>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="n">conn</span><span class="o">.</span><span class="n">upload_fild</span><span class="p">(</span><span class="n">file_id</span><span class="o">=</span><span class="s2">&quot;ENCFF852WVP&quot;</span><span class="p">,</span><span class="n">file_path</span><span class="o">=</span><span class="s2">&quot;/path/to/myfile&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="section" id="post-request">
<h2>POST Request<a class="headerlink" href="#post-request" title="Permalink to this headline">¶</a></h2>
<p>Let’s create a new File record on the Portal that represents a FASTQ file, and automatically upload
the file to AWS once that is done:</p>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="n">payload</span> <span class="o">=</span> <span class="p">{</span>
  <span class="s2">&quot;aliases&quot;</span><span class="p">:</span> <span class="p">[</span><span class="s2">&quot;michael-snyder:SCGPM_SReq-1103_HG7CL_L3_GGCTAC_R1.fastq.gz&quot;</span><span class="p">],</span>
  <span class="s2">&quot;dataset&quot;</span><span class="p">:</span> <span class="s2">&quot;ENCSR161EAA&quot;</span><span class="p">,</span>
  <span class="s2">&quot;file_format&quot;</span><span class="p">:</span> <span class="s2">&quot;fastq&quot;</span><span class="p">,</span>
  <span class="s2">&quot;flowcell_details&quot;</span><span class="p">:</span> <span class="p">{</span>
    <span class="s2">&quot;barcode&quot;</span><span class="p">:</span> <span class="s2">&quot;GGCTAC&quot;</span><span class="p">,</span>
    <span class="s2">&quot;flowcell&quot;</span><span class="p">:</span> <span class="s2">&quot;HG7CL&quot;</span><span class="p">,</span>
    <span class="s2">&quot;lane&quot;</span><span class="p">:</span> <span class="s2">&quot;3&quot;</span><span class="p">,</span>
    <span class="s2">&quot;machine&quot;</span><span class="p">:</span> <span class="s2">&quot;COOOPER&quot;</span>
  <span class="p">},</span>
  <span class="s2">&quot;output&quot;</span><span class="p">:</span> <span class="s2">&quot;reads&quot;</span><span class="p">,</span>
  <span class="s2">&quot;paired_end&quot;</span><span class="p">:</span> <span class="s2">&quot;1&quot;</span><span class="p">,</span>
  <span class="s2">&quot;platform&quot;</span><span class="p">:</span> <span class="s2">&quot;encode:HiSeq4000&quot;</span><span class="p">,</span>
  <span class="s2">&quot;read_length&quot;</span><span class="p">:</span> <span class="mi">101</span><span class="p">,</span>
  <span class="s2">&quot;replicate&quot;</span><span class="p">:</span> <span class="s2">&quot;michael-snyder:GM12878_eGFP-ZBTB11_CRISPR_ChIP_input_R1&quot;</span><span class="p">,</span>
  <span class="s2">&quot;submitted_file_name&quot;</span><span class="p">:</span> <span class="s2">&quot;/path/to/SCGPM_SReq-1103_HG7CL_L3_GGCTAC_R1.fastq.gz&quot;</span>
<span class="p">}</span>
</pre></div>
</div>
<p>Notice that we didn’t specify the required <cite>award</cite> and <cite>lab</cite> properties (required by the ENCODE
profiles). When not specified, the defaults will be taken from the environment variables
<cite>DCC_AWARD</cite> and <cite>DCC_LAB</cite> when present. Otherwise, you will get an error when trying to submit.
Before we can POST this though, we need to indicate the profile of the record-to-be.</p>
<div class="section" id="specifying-the-profile-key">
<h3>Specifying the profile key<a class="headerlink" href="#specifying-the-profile-key" title="Permalink to this headline">¶</a></h3>
<p>We are almost ready to hand this payload over to the <cite>post()</cite> method, however, we need to first
indicate the profile to POST to. To do this, add a special key to your payload that is stored in
the constant <cite>Connection.PROFILE_KEY</cite>.  The <cite>post()</cite> method depends on this key as the way of
indicating which profile to create a new record under. There are a few ways in which you can
specify the profile, but the recommended way is to use the stripped-down profile ID. If you
look at the JSON schema for the File profile at
<a class="reference external" href="https://www.encodeproject.org/profiles/file.json">https://www.encodeproject.org/profiles/file.json</a>, you’ll find that the value of it’s <cite>id</cite>
property is <cite>“/profiles/file.json”</cite>. The stripped-down value that you should use is <cite>file</cite>.
Another way to say it is to use the barebones profile name that you put in the URL to get to it.
See the documentation in the <cite>profile.Profile</cite> class for further details on how this works.</p>
<p>Without futher ado, let’s now add the profile specification to the payload and POST it:</p>
<div class="highlight-default"><div class="highlight"><pre><span></span><span class="n">payload</span><span class="p">[</span><span class="n">Connection</span><span class="o">.</span><span class="n">PROFILE_KEY</span><span class="p">]</span> <span class="o">=</span> <span class="s2">&quot;file&quot;</span>
<span class="n">conn</span><span class="o">.</span><span class="n">post</span><span class="p">(</span><span class="n">payload</span><span class="p">)</span>
</pre></div>
</div>
<p>The logging to STDOUT and your log files will indicate the progress of your request, including
the upload of your FASTQ file to AWS.</p>
</div>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper"><div class="relations">
<h3>Related Topics</h3>
<ul>
  <li><a href="index.html">Documentation overview</a><ul>
      <li>Previous: <a href="index.html" title="previous chapter">ENCODE Utils package</a></li>
      <li>Next: <a href="scripts/eu_register.html" title="next chapter">eu_register.py</a></li>
  </ul></li>
</ul>
</div>
<div id="searchbox" style="display: none" role="search">
  <h3>Quick search</h3>
    <form class="search" action="search.html" method="get">
      <div><input type="text" name="q" /></div>
      <div><input type="submit" value="Go" /></div>
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="footer">
      &copy;2018, Nathaniel Watson.
      
      |
      Powered by <a href="http://sphinx-doc.org/">Sphinx 1.6.6</a>
      &amp; <a href="https://github.com/bitprophet/alabaster">Alabaster 0.7.10</a>
      
      |
      <a href="_sources/examples.rst.txt"
          rel="nofollow">Page source</a>
    </div>

    

    
  </body>
</html>